azurelinuxagent/common/protocol/wire.py

```diff
@@ -614,8 +613,7 @@ def fetch_manifest(self, version_uris):
 
                 try:
                     host = self.get_host_plugin()
-                    uri, headers = host.get_artifact_request(version.uri)
-                    response = self.fetch(uri, headers, use_proxy=False)
+                    response = self.fetch(*host.get_artifact_request(version.uri), use_proxy=False)
```

Comment:
I prefer not to stack the function calls here, it reduces the clarity of stack traces.

Reply:
Fixed.